### PR TITLE
Escape variables in mkdocs data

### PR DESCRIPTION
**Pull request by davidfischer**

### CODE REVIEW

File: readthedocs/doc_builder/templates/doc_builder/data.js.tmpl [hunk 1]
```diff
@@ -1,10 +1,10 @@
 var READTHEDOCS_DATA = {{ data_json|safe }}
```
Comment (ericholscher) [inline readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: Do we not need it here, too?

Comment (davidfischer) [reply readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: Let me check. There's a [new templatetag](https://docs.djangoproject.com/en/2.2/ref/templates/builtins/#json-script) in Django 2 for JSON data but we can't use it yet.

Comment (stsewd) [reply readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: There is one in 1.11 https://docs.djangoproject.com/en/1.11/ref/templates/builtins/#escapejs

Comment (stsewd) [reply readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: oh, this is a json object I guess?

Comment (davidfischer) [reply readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: So the above is not quite right. This is a bit complex so I'm going to spend slightly more time here and try to write a test or two.

Comment (davidfischer) [reply readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: > oh, this is a json object I guess?

Correct. Escapejs doesn't work for this kind of object.

File: readthedocs/doc_builder/templates/doc_builder/data.js.tmpl [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
 
 // Old variables
-var doc_version = "{{ current_version }}";
-var doc_slug = "{{ slug }}";
-var page_name = "{{ pagename }}";
-var html_theme = "{{ html_theme }}";
+var doc_version = "{{ current_version|escapejs }}";
+var doc_slug = "{{ slug|escapejs }}";
+var page_name = "{{ pagename|escapejs }}";
+var html_theme = "{{ html_theme|escapejs }}";
```
Comment (stsewd) [inline readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: I think `"` are inserted by the filter.

File: readthedocs/core/templatetags/core_tags.py [hunk 3]
```diff
@@ -112,3 +112,28 @@ def key(d, key_name):
...
     return __version__
+
+
+@register.filter
+def escapejson(data, indent=None):
+    """
+    Escape JSON correctly for inclusion in Django templates
+
+    This code was mostly taken from Django's implementation
+    https://docs.djangoproject.com/en/2.2/ref/templates/builtins/#json-script
```
Comment (stsewd) [inline readthedocs/core/templatetags/core_tags.py]: Maybe we should link to the code too https://github.com/django/django/blob/2.2.2/django/utils/html.py#L74-L92

File: readthedocs/doc_builder/templates/doc_builder/data.js.tmpl (hunk 2 above)
Comment (stsewd) [reply readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: Reading the django code, looks like they don't include `"`

Comment (davidfischer) [reply readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: I'm not sure where you're looking but it is included as far as I can tell.

    >>> from django.template.defaultfilters import escapejs_filter
    >>> escapejs_filter('"')
    '\\u0022'

https://github.com/django/django/blob/1.11/django/utils/html.py#L54-L76

Comment (stsewd) [reply readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: I mean the `"` surrounding the variable. In jinja you just need to put `var foo = {{ bar|tojson }}`. Django needs  `var foo = "{{ bar|escapejs }"`

Comment (davidfischer) [reply readthedocs/doc_builder/templates/doc_builder/data.js.tmpl]: Got it. Correct. That's different.

**Pull request merged**